packages/react/src/TextInput/TextInput.tsx

```diff
@@ -137,6 +137,7 @@ const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
           visualPosition="leading"
           showLoadingIndicator={showLeadingLoadingIndicator}
           hasLoadingIndicator={typeof loading === 'boolean'}
+          data-testid="text-input-leading-visual"
```

Comment:
Heads up: We avoid adding our own `data-testid`. For tests, using aria-label or placeholder is recommended

Reply:
Oh interesting! I'm not sure I understand though, could you point me to an example? It seems strange to me to use an accessibility mechanism with specific semantics for this, especially when the component in question, `TextInputInnerVisualSlot`, is only used internally.

Reply:
Sorry to barge in - wanted to share this tiny resource in case it is helpful! https://testing-library.com/docs/queries/about#priority I like how they prioritise the queries.